### ISSUE TITLE
In high contrast mode, the "Explore Public Rooms" menu is not legible

### ISSUE THREAD

**Issue by garhow** [element-hq/element-web#34213]
### Steps to reproduce

1. Enable High Contrast mode in Appearance settings
2. Open the "Explore Public Rooms" menu

#### What did you expect?
I expected the High Contrast mode to increase contrast, not make the UI less readable. The text within the light gray boxes should be a dark color (preferably black), not a light color.

#### What happened instead?
The UI elements are light on light, which is unreadable.

### Operating system

macOS 27 Developer Beta 3

### Application version

Element version 1.12.23

### How did you install the app?

From https://element.io/en/download

### Homeserver

Dendrite 0.15.2

### Will you send logs?

Yes

<img width="968" height="859" alt="Image" src="https://github.com/user-attachments/assets/8ebe43c2-8048-4ca1-98c2-f161b17a428d" />
